graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/SchemaTransformer.java

```diff
@@ -142,8 +148,8 @@ private String sdl() {
         final SchemaPrinter.Options options = SchemaPrinter.Options.defaultOptions()
                 .includeScalarTypes(true)
                 .includeExtendedScalarTypes(true)
-                .includeSchemaDefintion(true)
-                .includeDirectives(true);
+                .includeSchemaDefinition(true)
+                .includeDirectives(directive -> !HIDDEN_DIRECTIVES.contains(directive.getName()));
```

Comment:
This will exclude the `@deprecated` directive. This directive is an important part of GraphQL APIs as this is how deprecations are communicated to API consumers. At Indeed our strategy for evolving schema relies heavily on this directive. Are you certain that this breaks composition of schemas at the gateway? If so, maybe it's best to fix this in the gateway?

Reply:
This will only exclude the directive definitions of `@deprecated`, `@include`, and `@skip` from the SDL.

It will not remove usages of the `@deprecated` directive. This is visible in `SchemaPrinter`'s source code at [https://github.com/graphql-java/graphql-java/blob/v14.0/src/main/java/graphql/schema/idl/SchemaPrinter.java#L715](https://github.com/graphql-java/graphql-java/blob/v14.0/src/main/java/graphql/schema/idl/SchemaPrinter.java#L715) and [https://github.com/graphql-java/graphql-java/blob/v14.0/src/main/java/graphql/schema/idl/SchemaPrinter.java#L681](https://github.com/graphql-java/graphql-java/blob/v14.0/src/main/java/graphql/schema/idl/SchemaPrinter.java#L681) .

To be safe, I've updated `FederationTest` to include an `@deprecated` directive on a type so we can know if `SchemaPrinter` changes its behavior in the future.

Apollo Gateway will add back the directive definitions for `@deprecated`, `@include`, and `@skip` when it composes the graphs of the implementing services.

Reply:
Ah yes that's correct, I was misled by the `includeDirectives` name.